cmake/modules/extensions.cmake

```diff
@@ -3335,6 +3335,46 @@ macro(zephyr_linker_memory_ifdef feature_toggle)
   endif()
 endmacro()
 
+# Usage:
+#   zephyr_linker_dts_section(PATH <path>)
+#
+# Zephyr linker devicetree memory section from path.
+#
+# This function specifies an output section for the platform in use based on its
+# devicetree configuration.
+#
+# The section will only be defined if the devicetree exists and has status okay.
+#
+# PATH <path>      : Devicetree node path.
+#
+function(zephyr_linker_dts_section)
+  set(single_args "PATH")
+  cmake_parse_arguments(DTS_SECTION "" "${single_args}" "" ${ARGN})
+
+  if(DTS_SECTION_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "zephyr_linker_dts_section(${ARGV0} ...) given unknown "
+	    "arguments: ${DTS_SECTION_UNPARSED_ARGUMENTS}"
+    )
+  endif()
+
+  if(NOT DEFINED DTS_SECTION_PATH)
+    return()
```

Comment:
`PATH` is a required argument, afaik.

So we should error out here.
For inspiration:
https://github.com/zephyrproject-rtos/zephyr/blob/5e79463ace625b649028e599e5ad75e52d3bc168/cmake/modules/extensions.cmake#L2588-L2589